src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -715,7 +715,8 @@ private void NotifyFeaturesAboutStateChange(DatabaseRecord record, long index)
                 Debug.Assert(string.Equals(Name, record.DatabaseName, StringComparison.OrdinalIgnoreCase),
                     $"{Name} != {record.DatabaseName}");
 
-                if (LastDatabaseRecordIndex >= index)
+                // index and LastDatabaseRecordIndex could have equal values when we transit from/to passive and want to update the tasks. 
+                if (LastDatabaseRecordIndex > index)
```

Comment:
What will cause us to _stop_ replicating to a  removed node?
The node is still in the topology, and I don't think the Loader will have any reason to remove it.